f3/granite.go

```diff
@@ -246,15 +254,20 @@ func (i *instance) receiveOne(msg *GMessage) {
 		round.prepared.Receive(msg.Sender, msg.Value)
 	case COMMIT:
 		round.committed.Receive(msg.Sender, msg.Value)
+	case DECIDE:
+		i.decision.Receive(msg.Sender, msg.Value)
 	default:
 		i.log("unexpected message %v", msg)
 	}
 
 	// Try to complete the current phase.
 	// Every COMMIT phase stays open to new messages even after the protocol moves on to
 	// a new round. Late-arriving COMMITS can still (must) cause a local decision, *in that round*.
+	// DECIDE messages are also independent of current phase or round.
 	if msg.Step == COMMIT {
 		i.tryCommit(msg.Round)
+	} else if msg.Step == DECIDE {
+		i.tryDecide(msg.Value)
```

Comment:
This case here, and the implementation of tryDecide, are mixing the Bracha amplification in with the core state transitions, which confuses things. Please separate them. Do the amplification from a separate place in this method, independent of the core state machine. I'm not convinced we will want/use the amplification if we switch to explicit evidence, but the rest of the logic should stay the same.

- Add `i.tryDecide` down at 289. It should have no parameter but follow the same pattern as `tryCommit`.
- Add a new `i.maybeAmplify(msg.Value)` call here if `msg.step == DECIDE` (but not an _else_ if - should still `tryCompletePhase()`).

Reply:
Done.